### PR TITLE
Amqp test

### LINKED ISSUE #5: Add test with AMQP transport

### PR DESCRIPTION
**Pull request by kunicmarko20**
closes #5

### CODE REVIEW

**Comment by kunicmarko20** [comment]
Big pain getting amqp extension to work on travis...

File: .travis.yml [hunk 1]
```diff
@@ -16,10 +19,34 @@ matrix:
...
+       wget http://ftp.debian.org/debian/pool/main/libr/librabbitmq/librabbitmq1_0.5.2-2_amd64.deb
+       sudo dpkg -i librabbitmq1_0.5.2-2_amd64.deb librabbitmq-dev_0.5.2-2_amd64.deb
+
+       local ext_name=amqp-1.9.3
+       local ext_so=amqp.so
+       local ext_dir=$(php -r "echo ini_get('extension_dir');")
+       local ext_cache=~/php-ext/$(basename $ext_dir)/$ext_name
+
+       if [[ -e $ext_cache/$ext_so ]]; then
+           echo extension = $ext_cache/$ext_so >> ~/.phpenv/versions/7.2/etc/conf.d/travis.ini
```
Comment (martin-georgiev) [inline .travis.yml]: Can `TRAVIS_PHP_VERSION` be used here so this path becomes dynamic and doesn't couple to the 7.2 version?

File: features/Context/EndToEndContext.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    /**
+     * @Then I should get a response
+     */
+    public function iShouldGetAResponse()
+    {
+        $receiver = $this->receiverLocator->get('amqp');
+
+        $receiver->receive(function (Envelope $envelope) use ($receiver)  {
```
Comment (martin-georgiev) [inline features/Context/EndToEndContext.php]:
```suggestion
        $receiver->receive(static function (Envelope $envelope) use ($receiver): void  {
```

**Comment by kunicmarko20** [comment]
should be good now

**Comment by martin-georgiev** [comment]
> should be good now

Thank you for doing it on Sunday :)

**Comment by kunicmarko20** [comment]
hah, np, I was online and wanted to finish it :smile:

**Pull request merged**